src/Domain/CoordinateMaps/Wedge3D.cpp

```diff
@@ -0,0 +1,328 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/Wedge3D.hpp"
+
+#include "DataStructures/MakeWithValue.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+namespace CoordinateMaps {
+
+Wedge3D::Wedge3D(const double inner_radius, const double outer_radius,
+                 const Direction<3> positioning_of_wedge,
+                 const double inner_sphericity)
+    : inner_radius_(inner_radius),
+      outer_radius_(outer_radius),
+      positioning_of_wedge_(positioning_of_wedge),
+      inner_sphericity_(inner_sphericity) {
+  ASSERT(inner_radius > 0, "The inner radius must be greater than zero.");
+  ASSERT(outer_radius > inner_radius,
+         "The outer radius must be larger than the inner radius.");
+  ASSERT(inner_sphericity >= 0 and inner_sphericity <= 1,
+         "Inner sphericity must be between 0 and 1");
+}
```

Comment:
`ASSERT(cube_radius > 0, "The cube radius must be greater than zero.");`

`ASSERT(sphere_radius > 0, "The sphere radius must be greater than zero.");`

`ASSERT(cube_radius < sphere_radius or cube_radius > sqrt(3.0) * sphere_radius,`
         `"The boundaries of the sphere and cube must not overlap.");`

Reply:
maybe "surfaces" instead of "boundaries"

Reply:
The overlap assertion should have some sphericity adjustment, shouldn't it?